mypy_json_report.py

```diff
@@ -12,15 +12,57 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import argparse
+import enum
 import json
 import sys
 from collections import Counter, defaultdict
 from dataclasses import dataclass
 from typing import Counter as CounterType, Dict, Iterator
 
 
+class ErrorCodes(enum.IntEnum):
+    DEPRECATED = 1
+
+
 def main() -> None:
-    print(produce_errors_report(sys.stdin))
+    """
+    The primary entrypoint of the program.
+
+    Parses the CLI flags, and delegates to other functions as appropriate.
+    For details of how to invoke the program, call it with `--help`.
+    """
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(title="subcommand")
+
+    parser.set_defaults(func=_no_command)
+
+    parse_parser = subparsers.add_parser(
+        "parse", help="Transform Mypy output into JSON."
+    )
+
+    parse_parser.set_defaults(func=_parse_command)
+
+    args = sys.argv[1:]
+    parsed = parser.parse_args(args)
```

Comment:
Is there any reason you do this work on `sys.argv` yourself? Argparse can do it by itself if you don't provide it with any parameters:
```suggestion
    parsed = parser.parse_args()
```

Reply:
Good call -- I didn't know that this was the default. I'll remove the custom stuff.